x/ccv/consumer/module.go

```diff
@@ -151,7 +151,18 @@ func (AppModule) ConsensusVersion() uint64 { return 1 }
 
 // BeginBlock implements the AppModule interface
 // Set the VSC ID for the subsequent block to the same value as the current block
+// Panic if the provider's channel was established and then closed
 func (am AppModule) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
+	channelID, found := am.keeper.GetProviderChannel(ctx)
+	if found && am.keeper.IsChannelClosed(ctx, channelID) {
+		// cleanup state
+		am.keeper.DeleteProviderChannel(ctx)
```

Comment:
It seems unnecessary here if we're just going to stop the consumer chain?

It will require off-chain coordination to restart the chain anyway. So we may as well leave it to off-chain coordination to decide what the new genesis should look like

Reply:
@AdityaSripal Yeah, but we should be cleaning the state as well as we can to facilitate the restart. Is there a reason not to do it?